network/src/peers/peers.rs

```diff
@@ -54,83 +59,67 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
             number_of_connecting_peers
         );
 
+        // Fetch the bootnodes.
+        let bootnodes = self.config.bootnodes();
+
         // Drop peers whose RTT is too high or have too many failures.
         for (addr, peer_quality) in self
             .peer_book
             .connected_peers()
             .iter()
             .map(|(addr, info)| (*addr, &info.quality))
         {
+            // Skip this check if the peer is a bootnode.
+            if bootnodes.contains(&addr) {
```

Comment:
nit: this could be a `filter` in the iterator above

Reply:
agreed, thanks!